File: clash_of_code/shortest/the-zimablue_donnors.py
a,b,c=[],[],[]
for i in range(int(input())):d,e,f=input().split();a.append(d);b.append(e);c.append(int(f))
d,e=input().split()
f,g=[],[]
for i in range(len(a)):
 if b[i]==d:f.append(a[i]);g.append(c[i])
for i in range(len(g)):g[i]=abs(g[i]-int(e))
if len(f):
 print("MOST COMPATIBLE DONOR :",f[g.index(min(g))])
 h=g.copy()
 h.sort()
 if len(f)>1:
  print("OTHER COMPATIBLE DONORS :")
  for i in h[1:]:
   print(f[g.index(i)])
else:
 if len(a):print("NO COMPATIBLE DONOR")
 else:print("NO DONOR AVAILABLE")

File: clash_of_code/shortest/the-zimablue_donnors.rb
n,*l,d=$<.map &:split
a,*b=l.select{_2==d[0]}.sort_by{(d[1].to_i-_3.to_i).abs}.map{_1[0]}
puts n[0]>?0?a.nil?? "NO COMPATIBLE DONOR":"MOST COMPATIBLE DONOR : #{a}#{b[0]?"
OTHER COMPATIBLE DONORS :
#{b*"\n"}":""}":"NO DONOR AVAILABLE"
